flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -416,56 +480,56 @@ public ResourceProfile subtract(ResourceProfile other) {
 
 		return new ResourceProfile(
 			subtractDoublesConsideringInf(cpuCores, other.cpuCores),
```

Comment:
Can we actually subtract a resource with some bigger field?
Like, we can get negative CPU which we do not check in the constructor anymore

Reply:
There's a `checkArgument` to make sure that `this` is larger than `other`, if this is what you are concerning.